http-client/Network/HTTP/Client/Request.hs

```diff
@@ -111,16 +113,30 @@ getUri req = URI
     , uriFragment = ""
     }
 
+applyAnyUriBasedAuth :: URI -> Request -> Request
+applyAnyUriBasedAuth uri =  applyBasicAuth (username authInfo) (password authInfo)
+ where
+   authInfo = uriUserInfo $ fromJust $ uriAuthority uri
```

Comment:
So @karun012 and myself were looking through this `network-uri` code, and it seems that uriAuthority will never be Nothing....  here is some code we came across:
````haskell
uri :: URIParser URI                                                       
uri =                                                                      
    do  { us <- try uscheme                                                
        -- ; ua <- option Nothing ( do { try (string "//") ; uauthority } )
        -- ; up <- upath                                                   
        ; (ua,up) <- hierPart                                              
        ; uq <- option "" ( do { _ <- char '?' ; uquery    } )             
        ; uf <- option "" ( do { _ <- char '#' ; ufragment } )             
        ; return $ URI                                                     
            { uriScheme    = us                                            
            , uriAuthority = ua                                            
            , uriPath      = up                                            
            , uriQuery     = uq                                            
            , uriFragment  = uf                                            
            }                                                              
        }                                                                  
...
hierPart :: URIParser ((Maybe URIAuth),String)
hierPart =                                    
        do  { _ <- try (string "//")          
            ; ua <- uauthority                
            ; up <- pathAbEmpty               
            ; return (ua,up)                  
            }                                 
    <|> do  { up <- pathAbs                   
            ; return (Nothing,up)             
            }                                 
    <|> do  { up <- pathRootLess              
            ; return (Nothing,up)             
            }                                 
    <|> do  { return (Nothing,"")             
            }                                 
...
uauthority :: URIParser (Maybe URIAuth) 
uauthority =                            
    do  { uu <- option "" (try userinfo)
        ; uh <- host                    
        ; up <- option "" port          
        ; return $ Just $ URIAuth       
            { uriUserInfo = uu          
            , uriRegName  = uh          
            , uriPort     = up          
            }                           
        }                               
````
So I could be wrong, but it looks like it is always going to result in a `Just`....?